packages/utils/src/utils/styled.tsx

```diff
@@ -67,29 +68,30 @@ export function styled<T extends keyof JSX.IntrinsicElements>(
     React.PropsWithoutRef<Props> & React.RefAttributes<HTMLElement>
   >;
 
-  const elements = Array.isArray(element) ? element : [element];
+  const elements = arrayify(element) as [
+    T,
+    ...((props: unknown) => JSX.Element)[]
+  ];
 
   /** The result "styled" component. */
   const Wrapped = React.forwardRef<HTMLElement, Props>((props, ref) => {
-    const { as,  ...rest } = props;
+    const { as, ...rest } = props;
```

Comment:
should this also do the same useEffect/useMemo stuff?

Reply:
Okay i think not because styled is only called once !

Think about it ... it created the function once and then u can pass an as prop to it but you can never update the original element list